postprocessing/pp-opus.c

```diff
@@ -72,30 +66,37 @@ int janus_pp_opus_process(FILE *file, janus_pp_frame_packet *list, int *working)
 	int bytes = 0, len = 0, steps = 0, last_seq = 0;
 	uint64_t pos = 0, nextPos = 0;
 	uint8_t *buffer = g_malloc0(1500);
+	AVPacket *pkt = av_packet_alloc();
+	AVRational timebase = {1, 48000};
+
 	while(*working && tmp != NULL) {
 		if(tmp->prev != NULL && ((tmp->ts - tmp->prev->ts)/48/20 > 1)) {
 			JANUS_LOG(LOG_WARN, "Lost a packet here? (got seq %"SCNu16" after %"SCNu16", time ~%"SCNu64"s)\n",
 				tmp->seq, tmp->prev->seq, (tmp->ts-list->ts)/48000);
-			ogg_packet *op = op_from_pkt((const unsigned char *)opus_silence, sizeof(opus_silence));
 			/* use ts differ to insert silence packet */
 			int silence_count = (tmp->ts - tmp->prev->ts)/48/20 - 1;
 			pos = (tmp->prev->ts - list->ts) / 48 / 20 + 1;
 			JANUS_LOG(LOG_WARN, "[FILL] pos: %06"SCNu64", writing silences (count=%d)\n", pos, silence_count);
 			int i=0;
 			for(i=0; i<silence_count; i++) {
-				pos = (tmp->prev->ts - list->ts) / 48 / 20 + i + 1;
+				pos = tmp->prev->ts - list->ts;
 				if(tmp->next != NULL)
-					nextPos = (tmp->next->ts - list->ts) / 48 / 20;
+					nextPos = tmp->next->ts - list->ts;
 				if(pos >= nextPos) {
-					JANUS_LOG(LOG_WARN, "[SKIP] pos: %06" SCNu64 ", skipping remaining silence\n", pos);
+					JANUS_LOG(LOG_WARN, "[SKIP] pos: %06" SCNu64 ", skipping remaining silence\n", pos / 48 / 20 + 1);
```

Comment:
Mh looking at these changes again (with fresh eyes from @atoppi) there seems to be something broken in the way `pos` is updated and handled now. I see how you decided to normalize both `pos` and `nextPos`, but before `pos` depended on `i`, while now it doesn't: since `tmp` never changes in the `for` loop, neither will the value of `pos`, making the checks broken. Considering `pos` is used as a basis for the `av_rescale_q` below, this would likely not insert all the silences we need, in case multiple packets were lost in a row.

Reply:
Oh, right. Updated to increase the pos by `20 * 48`.